subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformCachingIntegrationTest.groovy

```diff
@@ -697,8 +712,12 @@ class ArtifactTransformCachingIntegrationTest extends AbstractHttpDependencyReso
         output.count("Transformed") == 2
         isTransformed("dir1.classes", "dir1.classes.dir")
         isTransformed("lib1.jar", "lib1.jar.txt")
-        outputDir("dir1.classes", "dir1.classes.dir") != outputDir1
-        outputDir("lib1.jar", "lib1.jar.txt") != outputDir2
+        if (incrementalArtifactTransformations) {
+            projectOutputDir("dir1.classes", "dir1.classes.dir") == outputDir1
+        } else {
+            gradleUserHomeOutputDir("dir1.classes", "dir1.classes.dir") != outputDir1
+        }
```

Comment:
I wonder if we should test implementation details like the location of the workspace this directly. What are we trying to prove here?

Reply:
The test checks if the same workspace is used for incremental transformations.

Reply:
Yes, but we do it by making more assumptions about where it will be located.

Reply:
I removed the extra assumption and use `outputDir` in both cases.